### PR TITLE
Profile Management

### PR DESCRIPTION
**Pull request by ritika05**
Files Changed:
Startup.cs(Injected Dependency)
ApplicationUser.cs(Corrected Property Name of UpdateDateTime)
profile-dashboard.component.ts(Task No: 20860)
profile-dashboard.html(Task No: 20862)
profile-edit.component.ts(Task No: 20860)
profile-edit.html(Task No: 20862)
ProfileDetailsController.cs(Task No:20860)
IProfileRepository.cs(Task No: 20862)
ProfileRepository.cs(Task No: 20862)
Files Added:
profile.model.ts
profile.service.ts
Completed Tasks:
Task No: 20860 and 20862
Resolved all the issues :
created GetUserDetails method in ProfileRepository
removed name from UpdateUserDetails that was passed as parameter in ProfileDetailsController
explained what parameter is doing the comments section

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Core/Controllers/ProfileDetailsController.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using Promact.Trappist.Web.Models;
+using System.Threading.Tasks;
+
+namespace Promact.Trappist.Core.Controllers
+{
+  [Route("api/Profile")]
+  public class ProfileDetailsController : Controller
+  {
+    private readonly IProfileRepository _profileRepository;
+    private UserManager<ApplicationUser> _userManager;
```
Comment (iroshni) [inline Trappist/src/Promact.Trappist.Core/Controllers/ProfileDetailsController.cs]: why readonly is not written here ?

File: Trappist/src/Promact.Trappist.Repository/ProfileDetails/IProfileRepository.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace Promact.Trappist.Repository.ProfileDetails
+{
+    public interface IProfileRepository
+    {
+
+    /// <summary>
+    /// Updates current user's details
+    /// </summary>
+    /// <param name="updateUserDetails"></param>
```
Comment (iroshni) [inline Trappist/src/Promact.Trappist.Repository/ProfileDetails/IProfileRepository.cs]: No description written for parameters

File: Trappist/src/Promact.Trappist.Core/Controllers/ProfileDetailsController.cs (hunk 1 above)
Comment (ritika05) [inline Trappist/src/Promact.Trappist.Core/Controllers/ProfileDetailsController.cs]: Sorry Ma'am. I didn't give it. I'll add it

File: Trappist/src/Promact.Trappist.Core/Controllers/ProfileDetailsController.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    /// <summary>
+    /// Updates current user's details
+    /// </summary>
+    /// <param name="name"></param>
+    /// <param name="updateUserDetails"></param>
+    /// <returns>Update user details in database</returns>
+    [HttpPut("{name}")]
+    public IActionResult UpdateProfile([FromRoute]string name, [FromBody] ApplicationUser updateUserDetails)
```
Comment (iroshni) [inline Trappist/src/Promact.Trappist.Core/Controllers/ProfileDetailsController.cs]: what 'name' will do here ? as its not used anywhere in the method

**Pull request closed without merging**